app/controllers/application_controller.rb

```diff
@@ -5,6 +5,157 @@ class ApplicationController < Sinatra::Base
   configure do
     set :public_folder, 'public'
     set :views, 'app/views'
+    enable :sessions
+    set :session_secret, "fweets"
   end
 
-end
\ No newline at end of file
+  get '/' do
+    erb :index
+  end
+
+  get '/signup' do
+    if !logged_in?
+      erb :'/users/create_user'
+    else
+      redirect "/tweets"
+    end
+  end
+
+  post '/signup' do
+
+    user = User.new(username: params[:username], email: params[:email], password: params[:password])
+
+    if user.save
+      session[:user_id] = user.id
+      redirect "/tweets"
+    else
+      redirect "/signup"
+    end
+
+  end
+
+  get '/login' do
+    if !logged_in?
+      erb :'/users/login'
+    else
+      redirect "/tweets"
+    end
+  end
+
+  post '/login' do
+    user = User.find_by(username: params[:username])
+    if user && user.authenticate(params[:password])
+      session[:user_id] = user.id
+      redirect "/tweets"
+    else
+      redirect "/login"
+    end
+  end
+
+  get '/tweets' do
+    if logged_in?
+      @all_tweets = Tweet.all
+      erb :'/tweets/tweets'
+    else
+      redirect "/login"
+    end
+  end
+
+  get '/logout' do
+    if logged_in?
+      session.clear
+    end
+    redirect "/login"
+  end
+
+  get '/users/:slug' do
+    #if logged_in?
+      @user = User.find_by_slug(params[:slug])
+      @all_tweet = @user.tweets
+      erb :'/users/show'
+    #else
+    #  redirect "/login"
+    #end
+  end
+
+  get '/tweets/new' do
+    if logged_in?
+      erb :'/tweets/create_tweet'
+    else
+      redirect "/login"
+    end
+  end
+
+  post '/tweets' do
+    if !params[:content].empty?
+      tweet = Tweet.create(content: params[:content])
```

Comment:
The tweet creation and user association can both be accomplished in the same step:

``` ruby
@tweet = Tweet.create(content: params[:content], user: current_user)
```